checker/src/org/checkerframework/checker/nullness/KeyForAnnotatedTypeFactory.java

```diff
@@ -95,7 +96,11 @@ public KeyForAnnotatedTypeFactory(BaseTypeChecker checker) {
         Pair<Tree, AnnotatedTypeMirror> context = getVisitorState().getAssignmentContext();
 
         if (returnType.getKind() == TypeKind.DECLARED && context != null && context.first != null) {
-            AnnotatedTypeMirror assignedTo = TypeArgInferenceUtil.assignedTo(this, getPath(tree));
+            TreePath path = getPath(tree);
```

Comment:
How about putting this logic, which is duplicated in KeyForPropagationTreeAnnotator, into TypeArgInferenceUtil.assignedTo? If the path is null that method can just return null and then here things would behave the same.